torch_geometric/nn/models/label_usage.py

```diff
@@ -0,0 +1,92 @@
+import torch
+import torch.nn.functional as F
+
+from torch_geometric.typing import OptTensor, Adj
+
+class LabelUsage(torch.nn.Module):
+    r"""The label usage operator for semi-supervised node classification,
+    as introduced in `"Bag of Tricks for Node Classification"
+    <https://arxiv.org/abs/2103.13355>`_ paper.
+
+    .. note::
+
+        When using the :class:`LabelUsage`, adjust the model's input dimension
+        accordingly to include both features and classes.
+
+    Args:
+        base_model: An instance of the model that will do the 
+            inner forward pass.
+        num_classes (int): Number of classes in dataset
+        split_ratio (float): Proportion of true labels to use as features 
+            during training (default: :obj:'0.5')
+        num_recycling_iterations (int): Number of iterations for the
+            label reuse procedure to cycle predicted soft labels 
+            (default: :obj:'100')
+        return_tuple (bool): If true, returns (output, train_label_idx, 
```

Comment:
The paper impl only takes gradient steps on the `train_pred_idx` and thus we may want to make this true by default or even remove the option.  I haven't fully thought through if gradient steps on the `train_label_idx` might be harmful for different losses